joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstCreator.scala

```diff
@@ -50,14 +50,33 @@ class AstCreator(val relativeFileName: String, val parserResult: ParserResult, v
     memberAsts
   }
 
-  protected def astForMembers(members: Seq[DotNetNodeInfo]): Seq[Ast] = members.flatMap(astForNode)
-
+  protected def astForMembers(members: Seq[DotNetNodeInfo]): Seq[Ast] = members.flatMap(member => astForNode(member))
   protected def astForNode(json: Value): Seq[Ast] = {
     val nodeInfo = createDotNetNodeInfo(json)
     astForNode(nodeInfo)
   }
 
-  protected def astForNode(nodeInfo: DotNetNodeInfo): Seq[Ast] = {
+  protected def astForBlock(block: DotNetNodeInfo, order: Int = -1): Ast = {
+    val newBlockNode = blockNode(block).order(order).argumentIndex(order)
+    scope.pushNewScope(BlockScope)
+    var currOrder = 1
+    val childAsts = block
+      .json(ParserKeys.Statements)
+      .arrOpt
+      .getOrElse(List())
+      .arr
+      .flatMap { parserJsonValue =>
+        val parserNode = createDotNetNodeInfo(parserJsonValue)
```

Comment:
Doesn't the ordering happen automatically? Also, seems like we may conflict on `astForBlock` from this PR https://github.com/joernio/joern/pull/4059/files#diff-579dff9a498abfc9a590aab42fc2f1941d436ea09a7283a2c92836a26c06eda2R145